include/navigation/infomax.h

```diff
@@ -81,6 +88,42 @@ class InfoMax : public VisualNavigationBase
         return m_Weights;
     }
 
+    static MatrixType generateInitialWeights(const int numInputs,
+                                             const int numHidden,
+                                             const unsigned seed = std::random_device()())
+    {
+        // Note that we transpose this matrix after normalisation
+        MatrixType weights(numInputs, numHidden);
+
+        LOG_INFO << "Seed for weights is: " << seed;
+
+        std::default_random_engine generator(seed);
+        std::normal_distribution<FloatType> distribution;
+        std::generate_n(weights.data(), weights.size(), [&]() { return distribution(generator); });
+
+        LOG_VERBOSE << "Initial weights" << std::endl
+                    << weights;
+
+        // Normalise mean and SD for row so mean == 0 and SD == 1
+        const auto means = weights.rowwise().mean();
+        LOG_VERBOSE << "Means" << std::endl << means;
+
+        weights.colwise() -= means;
```

Comment:
Is normalising the columns by the rows (I _think_ that's what this does) intentional? with square matrices I think it'd work but probably not provide very good normalisation

Reply:
I did a double-take here because it does *look* wrong, but what it's actually doing is getting the mean for each column (as a vector which will be ``numColumns`` long) and then subtracting that from each of the columns. So in the end your columns end up having a mean of approx 0.